dashboard/src/shared/Auth.ts

```diff
@@ -44,6 +46,19 @@ export class Auth {
       }
     }
   }
+
+  // fetchOIDCToken does a HEAD request to collect the Bearer token
+  // from the authorization header if exists
+  public static async fetchOIDCToken(): Promise<string | null> {
+    const { headers } = await Axios.head("/");
+    if (headers && headers.authorization) {
+      const tokenMatch = (headers.authorization as string).match(/Bearer\s([a-zA-Z0-9\._-]*)/);
```

Comment:
Could you explain the change from `.*` to `[a-zA-Z0-9\._-]*`? Isn't it safer to use `.*` here, just in case we're missing permitting a character that a provider might use? For example, what if there is a padding and we need `=`? What if the token is not in JWT format (not sure if that's possible with OIDC, but I could, instead of using OIDC, also hardcode the Ingress with a specific bearer token my apiserver accepts to bypass all auth).

Reply:
Yes, I forgot the comment here. The problem is that when using the directive `add_header Authorization $http_authorization` with oauth2_proxy it's adding the Bearer token twice so the header looks like `Bearer ABC...XYZ, Bearer ABC...XYZ`. We can make it work just not adding that directive for oauth2_proxy but since it was simple enough I added the regex. In any case, it's true that it's not the safest regex, I can look for `,` instead.

Reply:
@andresmgot is it possible to disable that in oauth2_proxy by removing pass/set-authorization-token?